### ISSUE TITLE
ImGui scales instead of resizing on Wayland with GLFW and VULKAN

### ISSUE THREAD

**Issue by AndreiNego** [ocornut/imgui#7671]
### Version/Branch of Dear ImGui:

master

### Back-ends:

imgui_impl_glfw.cpp + imgui_impl_vulkan.cpp

### Compiler, OS:

Linux + clang

### Details:

Noticed that window resizing on Wayland results in stretching the ImGui windows offsetting the input as well.
Checking this revealed that err code VK_SUBOPTIMAL_KHR is never raised , this results in the swapchain never being rebuilt.

### Screenshots/Video:

![image](https://github.com/ocornut/imgui/assets/45206607/31db0194-fc59-448d-b928-0047b97672c7)

### Minimal, Complete and Verifiable Example code:

```cpp
// Here's some code anyone can copy and paste to reproduce your issue
if (err == VK_ERROR_OUT_OF_DATE_KHR || err == VK_SUBOPTIMAL_KHR)
    {
        g_SwapChainRebuild = true;
        return;
    }
```

**Comment by AndreiNego** [comment]
Added two callbacks in the main function to solve the issue, although just one should be enough.


glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
    GLFWwindow* window = glfwCreateWindow(1280, 720, "Dear ImGui GLFW+Vulkan example", nullptr, nullptr);
    if (!glfwVulkanSupported())
    {
        printf("GLFW: Vulkan Not Supported\n");
        return 1;
    }

    glfwSetWindowSizeCallback(window, [](GLFWwindow* window, int width, int height){g_SwapChainRebuild = true;});
    glfwSetFramebufferSizeCallback(window, [](GLFWwindow* window, int width, int height){g_SwapChainRebuild = true;});

**Comment by AndreiNego** [comment]
More info on the issue:
"(VK_SUBOPTIMAL_KHR happens only on X11/XCB or/and xWayland)
In Wayland resize logic completely different from other WM, and way too many platforms to support if you rly care about crossplatform. (and Wayland has too many features that I left untested, I have no idea how application will react to those WM features(exampel I was talking about “zoom” part of screen, what application do in this case how it rendered… no idea))"
https://community.khronos.org/t/vk-suboptimal-khr-is-it-safe-to-use-it-as-window-resize-detection/107848/5

**Comment by ocornut** [comment]
I am wondering if we could simply fix it by changing:
```cpp
        // Resize swap chain?
        if (g_SwapChainRebuild)
        {
            int width, height;
            glfwGetFramebufferSize(window, &width, &height);
            if (width > 0 && height > 0)
            {
                ImGui_ImplVulkan_SetMinImageCount(g_MinImageCount);
                ImGui_ImplVulkanH_CreateOrResizeWindow(g_Instance, g_PhysicalDevice, g_Device, &g_MainWindowData, g_QueueFamily, g_Allocator, width, height, g_MinImageCount);
                g_MainWindowData.FrameIndex = 0;
                g_SwapChainRebuild = false;
            }
        }
```
to
```cpp
        // Resize swap chain?
        int fb_width, fb_height;
        glfwGetFramebufferSize(window, &fb_width, &fb_height);
        if (g_SwapChainRebuild || g_MainWindowData.Width != fb_width || g_MainWindowData.Height != fb_height)
            if (fb_width > 0 && fb_height > 0)
            {
                ImGui_ImplVulkan_SetMinImageCount(g_MinImageCount);
                ImGui_ImplVulkanH_CreateOrResizeWindow(g_Instance, g_PhysicalDevice, g_Device, &g_MainWindowData, g_QueueFamily, g_Allocator, fb_width, fb_height, g_MinImageCount);
                g_MainWindowData.FrameIndex = 0;
                g_SwapChainRebuild = false;
            }
```
Could you test it?

Can you already see if things works correctly on your system with the SDL+Vulkan example?

Thank you!

**Comment by AndreiNego** [comment]
Hi,
I've tested it and it worked perfectly.
Unfortunately, for SDL I am unable to get SDL.h to install on my system. I have two potential packages that provide the header but none of them actually install it.

**Comment by ocornut** [comment]
Pushed fixes as a47bfb1 and 7538ca6
Looking at equivalent SDL example it seemed clear the same thing would happen.
If you get around to install SDL please verify it before and after 🙏

**Comment by AndreiNego** [comment]
For SDL the problem does not exist.
SDL does not open a Wayland native window but an X window.
The flag is raised for VK_ERROR_OUT_OF_DATE_KHR during resize
![image](https://github.com/ocornut/imgui/assets/45206607/72f7ddf4-ffde-4b11-a397-37e000f716e6)
